tidb/src/main/java/io/tidb/bigdata/tidb/buffer/DeduplicateRowBuffer.java

```diff
@@ -63,11 +65,18 @@ public boolean add(Row row) {
               indexColumns.stream()
                   .map(i -> row.get(i.getOffset(), null))
                   .collect(Collectors.toList()));
+
+      // Since NULL != NULL, we don't deduplicate key with NULL.
+      if (indexValue.contains(null)) {
+        rows.add(row);
+        return true;
+      }
```

Comment:
we need to continue rather than add row and return true.
because indexValue.contains(null) only shows this uniqueIndex has no conflict, we also need to check another uniqueIndex in next loop